presto-spi/src/main/java/com/facebook/presto/spi/plan/TableScanNode.java

```diff
@@ -101,6 +106,15 @@ public TableHandle getTable()
         return assignments;
     }
 
+    /**
+     * Get whether this is a tableScan for deletion
+     */
+    @JsonProperty
+    public boolean getIsDelete()
```

Comment:
nit: `getIsDelete` looks weird. The Java convention for boolean geters is to start with `is*`

How about renaming the geter to `isDelete`, and renaming the field to simply `delete`?

Reply:
I originally called the getter `isDelete, but the json serialization didn't work unless the getter was called `getXXXX`

Reply:
Make sure you specify the field name correctly in the constructor:

`@JsonProperty("getIsDelete") boolean ...` -> `@JsonProperty("delete") boolean ...`.

Unless the JSON framework specifically doesn't like the `delete` word, it should work. See : https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java#L53

Reply:
maybe it was because both the field and getter were called `isDelete`.  Anyway, I'll try out your suggestion.

Reply:
Oh, yeah. If you call both `isDelete` it wont work. The field name should be without `is`